cmd/goal/asset.go

```diff
@@ -576,17 +578,15 @@ var freezeAssetCmd = &cobra.Command{
 	},
 }
 
-func assetDecimalsFmt(amount uint64, decimals uint32) string {
+func assetDecimalsFmt(amount uint64, decimals uint64) string {
 	// Just return the raw amount with no decimal if decimals is 0
 	if decimals == 0 {
 		return fmt.Sprintf("%d", amount)
 	}
 
 	// Otherwise, ensure there are decimals digits to the right of the decimal point
-	pow := uint64(1)
-	for i := uint32(0); i < decimals; i++ {
-		pow *= 10
-	}
+	pow := uint64(math.Pow(10, float64(decimals)))
+
```

Comment:
This makes me nervous.  `math.Pow(10, float64(decimals)` could be *slightly* less than the exact integer due to floating point representation, and then the uint64 cast will bring it down, not round.  I'd rather we keep the integral exponentiation.